File: web/static/js/mission_control.js
function loadMission() {
    const today = new Date().toISOString().slice(0, 10);

    fetch(`/api/mission/load/${today}`)
        .then(r => r.json())
        .then(data => {
            renderHealth(data);
            renderCharts(data);
            renderEvents(data);
        });
}

function renderHealth(data) {
    const el = document.getElementById("health-summary");

    const conf = data.timing_confidence?.score ?? 0;
    const stab = data.stability_prediction?.predicted_timing_error_ns ?? 0;
    const recv = data.receiver_health?.issues ?? ["unknown"];
    const env = data.environment_change?.status ?? "unknown";
    const sla = data.sla?.overall_compliant ? "OK" : "VIOLATION";

    el.innerHTML = `
        <p><strong>Timing Confidence:</strong> ${conf}</p>
        <p><strong>Predicted Stability:</strong> ${stab.toFixed(1)} ns</p>
        <p><strong>Receiver Health:</strong> ${recv.join(", ")}</p>
        <p><strong>Environment:</strong> ${env}</p>
        <p><strong>SLA:</strong> ${sla}</p>
    `;
}

function renderCharts(data) {
    Plotly.newPlot("confidence-chart", [{
        x: data.charts.confidence.map(p => new Date(p.t * 1000)),
        y: data.charts.confidence.map(p => p.v),
        mode: "lines"
    }]);

    Plotly.newPlot("error-chart", [{
        x: data.charts.timing_error.map(p => new Date(p.t * 1000)),
        y: data.charts.timing_error.map(p => p.v),
        mode: "lines"
    }]);

    Plotly.newPlot("interference-chart", [{
        x: data.charts.interference.map(p => new Date(p.t * 1000)),
        y: data.charts.interference.map(p => p.v),
        mode: "lines"
    }]);

    Plotly.newPlot("availability-chart", [{
        x: data.charts.availability.map(p => new Date(p.t * 1000)),
        y: data.charts.availability.map(p => p.v),
        mode: "lines"
    }]);
}

function renderEvents(data) {
    const feed = document.getElementById("event-feed");

    const events = [
        ...data.events.gnss,
        ...data.events.ptp,
        ...data.events.interference,
        ...data.events.sla
    ].sort((a, b) => a.timestamp - b.timestamp);

    feed.textContent = events
        .slice(-50)
        .map(e => {
            const ts = new Date(e.timestamp * 1000).toISOString();
            return `${ts} — ${e.type}`;
        })
        .join("\n");
}

loadMission();

File: backend/mission_control.py
from flask import Blueprint, jsonify
import os, json

mission_api = Blueprint("mission_api", __name__)

@mission_api.route("/load/<day>")
def load_mission(day):
    base = f"/opt/ptp-data/live/{day}"

    def load(name, default=None):
        file = os.path.join(base, name)
        if os.path.exists(file):
            return json.load(open(file))
        return default

    return jsonify({
        "timing_confidence": load("timing_confidence.json", {}),
        "stability_prediction": load("stability_prediction.json", {}),
        "receiver_health": load("receiver_health.json", {}),
        "environment_change": load("environment_change.json", {}),
        "sla": load("sla_report.json", {}),
        "gnss_metrics": load("gnss_metrics.json", {}),
        "ptp_metrics": load("ptp_metrics.json", {}),
        "events": {
            "gnss": load("gnss_events.json", []),
            "ptp": load("ptp_events.json", []),
            "interference": load("interference.json", []),
            "sla": load("sla_violations.json", [])
        },
        "charts": {
            "confidence": load("confidence_timeseries.json", []),
            "timing_error": load("timing_error_timeseries.json", []),
            "interference": load("interference_timeseries.json", []),
            "availability": load("availability_timeseries.json", [])
        }
    })
